### PR TITLE
allow dash "-" in resource name

### PR DESCRIPTION
**Pull request by liuyuan10**
according to
https://github.com/kubernetes/kubernetes/blob/b1f901acf44f8adf69759db434690c2a8bdd616b/staging/src/k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1/api.proto#L41, resource name is expected to be a DNS lable so dash should be allowed

Fixes: #476

### CODE REVIEW

**Comment by liuyuan10** [comment]
@adrianchiris

**Review by adrianchiris — APPROVED** [review]
LGTM

**Comment by adrianchiris** [comment]
need to sort UT in `manager_test.go`

**Comment by liuyuan10** [comment]
> just need to sort UT in `manager_test.go` to make CI happy

done. Thanks for pointing it out.

**Review by adrianchiris — APPROVED** [review]

**Comment by liuyuan10** [comment]
@adrianchiris do we need anything else before merging this?

**Comment by adrianchiris** [comment]
you need additional LGTM. already asked the folks in yesterday's meeting.

**Review by Eoghan1232 — APPROVED** [review]

**Comment by Eoghan1232** [comment]
lgtm

**Pull request merged**
